hnn_core/network.py

```diff
@@ -1170,6 +1196,74 @@ def __repr__(self):
 
         return entr
 
+    def drop(self, probability):
+        """Remove/keep a random subset of connections.
+
+        Parameters
+        ----------
+        probability : float
+            Probability of connection between any src-target pair.
+            Defaults to 1.0 producing an all-to-all pattern.
+
+        Notes
+        -----
+        num_srcs and num_targets are not updated after pruning connections.
+        These variables are meant to describe the set of original connections
+        before they are randomly removed.
+
+        The probability attribute will store the most recent value passed to
+        this function. As such, this number does not accurately describe the
+        connections probability of the original set after successive calls.
```

Comment:
keep the complexity bounded, otherwise software will become unmaintainable. You should try to address 80 percent usecases. Look up [feature creep](https://en.wikipedia.org/wiki/Feature_creep). I'd suggest throwing an error if someone tries to drop connections more than once